File: Codechef/CONI2015/CN05.cpp
// False Fibonacci Series

#include <stdio.h>
#include <math.h>

#define MOD 1000000007

long long modpow(int base, int exp, int modulus) {
  base %= modulus;
  unsigned long long  result = 1;
  while (exp > 0) {
          if (exp & 1) result = (result * base) % modulus;
          base = (base * base) % modulus;
          exp >>= 1;
        }
  return result;
}

int main() {
  unsigned long long* series = new unsigned long long[100000001];

  int T; scanf("%d", &T);
  while(T--){
    int N; scanf("%d", &N);
    long long ans = 0;
    if(N == 0) ans = 2;
    else if(N == 1) ans = 5;
    else {
      N--;
      ans = modpow(3, N, MOD) * 5;
      ans %= MOD;
    }
    printf("%llu\n", ans);
  }
}

File: Codechef/CONI2015/CN05.py
# False Fibonacci Series

import math

T = input()

while T:
    N = input()
    if N == 0:
        print 2
    elif N == 1:
        print 5
    else:
        N -= 1;
        ans = 3**N * 5 + (3**N-(-1)**N)/4
        ans = ans % 1000000007
        print ans

    T -= 1
